locking/models.py

```diff
@@ -50,21 +50,30 @@ def acquire_lock(self, obj=None, max_age=None, lock_name=''):
         if obj is not None:
             lock_name = _get_lock_name(obj)
 
-        try:
-            lock, created = self.get_or_create(locked_object=lock_name,
-                                               defaults={'max_age': max_age})
-            if not created:
-                # check whether lock is expired
-                if lock.is_expired:
-                    # Create a new lock to provide a new id for renewal.
-                    # This ensures the owner of the previous lock doesn't
-                    # remain in possession of the active lock id.
-                    lock.release()
-                    lock = self.create(locked_object=lock_name, max_age=max_age)
-                else:
-                    raise AlreadyLocked()
-        except IntegrityError:
-            raise AlreadyLocked()
+        with transaction.atomic():
```

Comment:
As I mentioned in the the PR from a couple of years ago, I don't think this atomic block provides much value. From what I can tell it's already in production, so if there haven't been any issues (such as deadlocks) I'm happy to leave as-is.